engines/startrek/startrek.cpp

```diff
@@ -0,0 +1,2229 @@
+		return SharedPtr<FileStream>(new FileStream(stream, bigEndian));
+	} else {
+		if (fileCount != 1) {
+			dataRunFile->seek(indexOffset);
+
+			indexOffset = dataRunFile->readByte() + (dataRunFile->readByte() << 8) + (dataRunFile->readByte() << 16);
+			//indexOffset &= 0xFFFFFE;
+
+			for (uint16 i = 0; i < fileIndex; i++) {
+				uint16 size = dataRunFile->readUint16LE();
+				indexOffset += size;
+			}
+		}
+		dataFile->seek(indexOffset);
+
+		uncompressedSize = (getPlatform() == Common::kPlatformAmiga) ? dataFile->readUint16BE() : dataFile->readUint16LE();
+		uint16 compressedSize = (getPlatform() == Common::kPlatformAmiga) ? dataFile->readUint16BE() : dataFile->readUint16LE();
+
+		Common::SeekableReadStream *stream = decodeLZSS(dataFile->readStream(compressedSize), uncompressedSize);
+
+		delete dataFile;
+		delete dataRunFile;
+		return SharedPtr<FileStream>(new FileStream(stream, bigEndian));
+	}
+
+	// We should not get to this point...
+	error("Could not find data for \'%s\'", filename.c_str());
+
+	return SharedPtr<FileStream>();
+}
+
+SharedPtr<FileStream> StarTrekEngine::loadFileWithParams(Common::String filename, bool unk1, bool unk2, bool unk3) {
+	return loadFile(filename);
+}
+
+void StarTrekEngine::playMovie(Common::String filename) {
+	if (getPlatform() == Common::kPlatformMacintosh)
+		playMovieMac(filename);
+	else
+		error("Interplay MVE not yet supported");
+}
+
+void StarTrekEngine::playMovieMac(Common::String filename) {
+	// Swap to 16bpp mode
+	initGraphics(512, 384, NULL);
+
+	Video::QuickTimeDecoder *qtDecoder = new Video::QuickTimeDecoder();
+
+	if (!qtDecoder->loadFile(filename))
+		error("Could not open '%s'", filename.c_str());
+
+	bool continuePlaying = true;
+
+	qtDecoder->start();
+
+	while (!qtDecoder->endOfVideo() && !shouldQuit() && continuePlaying) {
+		if (qtDecoder->needsUpdate()) {
+			const ::Graphics::Surface *frame = qtDecoder->decodeNextFrame();
+
+			if (frame) {
+				::Graphics::Surface *convertedFrame = frame->convertTo(_system->getScreenFormat());
+				_system->copyRectToScreen(convertedFrame->getPixels(), convertedFrame->pitch, 0, 0, convertedFrame->w, convertedFrame->h);
+				_system->updateScreen();
+				convertedFrame->free();
+				delete convertedFrame;
+			}
+		}
+
+		Common::Event event;
+		while (g_system->getEventManager()->pollEvent(event))
+			if (event.type == Common::EVENT_KEYDOWN && event.kbd.keycode == Common::KEYCODE_ESCAPE)
+				continuePlaying = false;
+
+		g_system->delayMillis(10);
+	}
+
+	delete qtDecoder;
+
+	// Swap back to 8bpp mode
+	initGraphics(SCREEN_WIDTH, SCREEN_HEIGHT);
+}
+
+uint16 StarTrekEngine::getRandomWord() {
+	return _randomSource.getRandomNumber(0xffff);
+}
+
+Common::String StarTrekEngine::getLoadedText(int textIndex) {
+	SharedPtr<FileStream> txtFile = loadFile(_txtFilename + ".txt");
+
+	byte *data = txtFile->_data;
+	while (textIndex != 0) {
+		while (*(data++) != '\0');
+		textIndex--;
+	}
+
+	return (char *)data;
+}
+
+} // End of namespace StarTrek
```

Comment:
This file is quite big. I recommend to move the methods to different files (leaving them in StarTreckEngine class): intro.cpp, inventory.cpp, animation.cpp, actors.cpp, or whatever else makes sense.